core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -4201,8 +4197,8 @@ public void completeTx(SendRequest req) throws InsufficientMoneyException {
             checkArgument(!req.completed, "Given SendRequest has already been completed.");
             // Calculate the amount of value we need to import.
             Coin value = Coin.ZERO;
-            boolean requiresInputs = req.tx.getType() != Transaction.Type.TRANSACTION_SUBTX_RESETKEY;
-            boolean requiresNoFee = req.tx.getType() == Transaction.Type.TRANSACTION_SUBTX_RESETKEY;
+            boolean requiresInputs = true;
```

Comment:
Can't this variable be removed? By reading the method it looks like its value never changes code execution.

Reply:
It can, but I did find one transaction type that does not require fees or inputs, which is for quorum commitments.  Therefore I will leave this code in for now, but change it to:

```
boolean requiresInputs = req.tx.getType() != Transaction.Type.TRANSACTION_QUORUM_COMMITMENT;
boolean requiresNoFee = req.tx.getType() == Transaction.Type.TRANSACTION_QUORUM_COMMITMENT;
```